src/System.Management.Automation/engine/parser/SemanticChecks.cs

```diff
@@ -494,20 +493,30 @@ public override AstVisitAction VisitForEachStatement(ForEachStatementAst forEach
             // Parallel flag not allowed
             if ((forEachStatementAst.Flags & ForEachFlags.Parallel) == ForEachFlags.Parallel)
             {
-                bool reportError = !forEachStatementAst.IsInWorkflow();
-                if (reportError)
-                {
-                    _parser.ReportError(forEachStatementAst.Extent,
-                        nameof(ParserStrings.ParallelNotSupported),
-                        ParserStrings.ParallelNotSupported);
-                }
+                _parser.ReportError(
+                    forEachStatementAst.Extent,
+                    nameof(ParserStrings.KeywordParameterReservedForFutureUse),
+                    ParserStrings.KeywordParameterReservedForFutureUse,
+                    "foreach",
+                    "parallel");
+            }
+
+            if (forEachStatementAst.ThrottleLimit != null)
+            {
+                _parser.ReportError(
+                    forEachStatementAst.Extent,
+                    nameof(ParserStrings.KeywordParameterReservedForFutureUse),
+                    ParserStrings.KeywordParameterReservedForFutureUse,
+                    "foreach",
+                    "throttlelimit");
             }
 
             // Throttle limit must be combined with Parallel flag
             if ((forEachStatementAst.ThrottleLimit != null) &&
                 ((forEachStatementAst.Flags & ForEachFlags.Parallel) != ForEachFlags.Parallel))
             {
-                _parser.ReportError(forEachStatementAst.Extent,
+                _parser.ReportError(
+                    forEachStatementAst.Extent,
                     nameof(ParserStrings.ThrottleLimitRequiresParallelFlag),
                     ParserStrings.ThrottleLimitRequiresParallelFlag);
```

Comment:
Why do we keep the error message if we already report both "parallel" and "throttlelimit"?

Reply:
I left that error in place so that it could be used later. It is not likely to come up, so I think that's ok, but if we want it removed, I can remove it.

Reply:
I can’t come up with a scenario where it would be useful.

Reply:
Maybe I'm mistaken, but the reason they wanted `-parallel` kept is for future use, so that we could have a `foreach -parallel` much like we're building a cmdlet version of the same right now. That is why I kept it, for future use (and to avoid someone coming along and saying "put that back", which has happened a lot with workflow deprecation, so I'm taking a more conservative approach).

Reply:
I understand that you say about future implementation of `foreach -parallel -throttlelimit` - in the case I agree to keep this.